### PR TITLE
[TSD-59] Keep existing tags, fields as it is after processing the ticket

### PR DESCRIPTION
**Pull request by HirotoShioi**
https://iohk.myjetbrains.com/youtrack/issue/TSD-59

Description:
For some reason, Zendesk changes the values of the custom field to `null` when posting the comment unless we explicitly tell Zendesk not to change it.
Also, we've been overwriting the existing tags.

This PR will fix these issues.

### CODE REVIEW

**Comment by HirotoShioi** [comment]
I've tested and it works.
I've tried adding other fields (`C.C`, `Type`) to see if the classifier removes/changes it but it didn't.
If we were to make changes to the ticket information, we should be careful about it..!

File: src/DataSource/DB.hs [hunk 1]
```diff
@@ -46,6 +47,17 @@ cachedZendeskLayer = ZendeskLayer
...
+instance FromField TicketFieldId where
+    fromField (Field (SQLInteger tfId) _)     = Ok . TicketFieldId . fromIntegral $ tfId
+    fromField f                             = returnError ConversionFailed f "need a text, ticket status"
+
+instance FromField TicketFieldValue where
+    fromField (Field (SQLText tfValue) _)     = Ok . TicketFieldValue $ tfValue
+    fromField f                             = returnError ConversionFailed f "need a text, ticket status"
+
+instance FromField [TicketField] where
+    fromField _ = undefined
```
Comment (ksaric) [inline src/DataSource/DB.hs]: ?? There should be a warning, try fixing this.

File: src/DataSource/Http.hs [hunk 2]
```diff
@@ -159,24 +162,33 @@ iteratePages req = do
...
-                   )
-                   req1
-    void $ liftIO $ apiCall (pure . encodeToLazyText) req2
+    let responseTicket = createResponseTicket (cfgAgentId cfg) ticketInfo zendeskResponse
+    let url  = showURL $ TicketsURL (zrTicketId zendeskResponse)
+    let req = addJsonBody responseTicket (apiRequest cfg url)
+    void $ liftIO $ apiCall (pure . encodeToLazyText) req
+
+-- | Create response ticket
+createResponseTicket :: Integer -> TicketInfo -> ZendeskResponse -> Ticket
```
Comment (ksaric) [inline src/DataSource/Http.hs]: Nice.

File: src/DataSource/Types.hs [hunk 3]
```diff
@@ -261,9 +265,9 @@ data RequestType
 data ZendeskResponse = ZendeskResponse
     { zrTicketId :: !TicketId
     , zrComment  :: !Text
-    , zrTags     :: ![Text] -- TODO(ks): This should be wrapped
+    , zrTags     :: !TicketTags
```
Comment (ksaric) [inline src/DataSource/Types.hs]: It's really regarding the previous PR, but good :+1:

File: src/DataSource/Types.hs [hunk 4]
```diff
@@ -428,6 +463,8 @@ instance Arbitrary TicketInfo where
             , tiUrl         = ticketUrl
             , tiTags        = ticketTags
             , tiStatus      = ticketStatus
+            , tiField       = ticketField
+            , tiCustomField  = ticketCustomField
```
Comment (ksaric) [inline src/DataSource/Types.hs]: Space :telescope:

File: src/DataSource/Types.hs [hunk 5]
```diff
@@ -294,12 +298,27 @@ newtype CommentOuter = CommentOuter {
 
 -- | Zendesk ticket
 data Ticket = Ticket
-    { tComment :: !Comment
+    { tComment     :: !Comment
     -- ^ Ticket comment
-    , tTag     :: ![Text]
+    , tTag         :: !TicketTags
+    , tField       :: ![TicketField]
```
Comment (ksaric) [inline src/DataSource/Types.hs]: I'm not entirely convinced that we need this. Let's have a chat first.

File: src/DataSource/DB.hs (hunk 1 above)
Comment (HirotoShioi) [reply src/DataSource/DB.hs]: Yes I will! Thank you!

File: src/DataSource/Types.hs (hunk 5 above)
Comment (HirotoShioi) [reply src/DataSource/Types.hs]: Sure!

**Comment by HirotoShioi** [comment]
@ksaric Thank you for checking!
I've resolved merge conflict, fix the format of the log.
If there's not problem, I can merge it!

**Pull request merged**